lib/datadog/tracing/metadata/tagging.rb

```diff
@@ -65,6 +65,18 @@ def set_tags(tags)
           tags.each { |k, v| set_tag(k, v) }
         end
 
+        # Returns true if the provided `tag` was set to a non-nil value.
+        # False otherwise.
+        #
+        # DEV: This method enables short-hand assertions on span tags:
+        # DEV: `expect(spans).to have_tag('http.method')`
+        #
+        # @param [String] tag the tag or metric to check for presence
+        # @return [Boolean] if the tag is present and not nil
+        def tag?(tag)
+          !get_tag(tag).nil? # nil is considered not present, thus we can't use `Hash#has_key?`
```

Comment:
Is there a slightly more direct/performant way of doing this? Access the `@` var itself? Or do we need to manage both the `meta` and `metric` vars?

On the fence about extending the API like this. I guess it's harmless, but it feels like adding in a test abstraction like this into production code seems not quite right. "Leaky." If there's any legit use of this outside the test suite, I think this is fine.

Otherwise, I think it would be wise to consider implementing such abstractions from within the test suite, if at all possible.